crawl-ref/source/game-options.cc

```diff
@@ -7,9 +7,38 @@
 
 #include "game-options.h"
 #include "options.h"
+#include "menu.h"
+#include "message.h"
 #include "misc.h"
+#include "prompt.h"
+#include "scroller.h"
 #include "tiles-build-specific.h"
 
+// Return a list of the valid field values for _curses_attribute(), excluding
+// hi: and hilite: values, which are all duplicates.
+// This is called after initialisation to ensure that colour_to_str() works.
+static map<unsigned, string>& _curses_attribute_map()
+{
+    static map<unsigned, string> list;
+    if (list.empty())
+    {
+        // This is the same list as in _curses_attribute.
+        list =
+        {
+            {CHATTR_NORMAL, "none"}, {CHATTR_STANDOUT, "standout"},
+            {CHATTR_BOLD, "bold"}, {CHATTR_BLINK, "blink"},
+            {CHATTR_UNDERLINE, "underline"}, {CHATTR_REVERSE, "reverse"},
+            {CHATTR_DIM, "dim"}
```

Comment:
Probably more readable if there's one line per entry, rather than try to cram multiple `CHATTR_` entries onto one line.